### PR TITLE
fix: react-router-dom bump

### LINKED ISSUE #4098: Bumping react-router-dom causes property errors
**Describe the bug**
When bumping `react-router-dom` from 5.0.1 to 5.1.1 causes the following errors:
image
image

**To Reproduce**

1. Go to `package.json` and bump `react-router-dom` version
2. Run `npm install`
3. Run `npm run test:unit`
4. See errors

**Expected behavior**
No errors.

**Additional context**
Think it is something to do with the test helpers work recently: https://github.com/bbc/simorgh/pull/3945

### PR DESCRIPTION
**Pull request by mykeels**
Resolves https://github.com/bbc/simorgh/issues/4098

**Overall change:** _Fixes test errors caused by react-router-dom bump._

**Code changes:**

- _Modify `App.jsx` to import `withRouter` from `react-router`_
- _Use `jest.mock(...)` syntax to ensure the code can mock, without attempting to directly `set` `ReactRouter.StaticRouter`_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Review by oluoluoxenfree — APPROVED** [review]

File: package-lock.json [hunk 1]
```diff
@@ -17573,15 +17573,15 @@
       }
     },
     "react-router-dom": {
-      "version": "5.0.1",
-      "resolved": "https://registry.npmjs.org/react-router-dom/-/react-router-dom-5.0.1.tgz",
-      "integrity": "sha512-zaVHSy7NN0G91/Bz9GD4owex5+eop+KvgbxXsP/O+iW1/Ln+BrJ8QiIR5a6xNPtrdTvLkxqlDClx13QO1uB8CA==",
+      "version": "5.1.2",
```
Comment (EinsteinNjoroge) [inline package-lock.json:17576]: Wondering why this changed to `5.1.2` while package.json has version `5.1.1`

File: package.json [hunk 2]
```diff
@@ -107,7 +107,7 @@
     "react-helmet": "^5.2.1",
     "react-lazyload": "^2.6.2",
     "react-router-config": "^5.1.1",
-    "react-router-dom": "^5.0.1",
+    "react-router-dom": "^5.1.1",
```
Comment (ryanmccombe) [inline package.json]:
```suggestion
    "react-router-dom": "^5.1.2",
```

**Review by ryanmccombe — APPROVED** [review]

File: src/app/containers/App/index.test.jsx [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
 import { render } from '@testing-library/react';
 import '@testing-library/jest-dom/extend-expect';
-import ReactRouter from 'react-router-dom';
+import * as ReactDOMRouter from 'react-router-dom';
 import { ClientApp, ServerApp } from '.';
 import * as App from './App';
 
 jest.mock('./App', () => jest.fn(() => <>Mocked App component</>));
 
+jest.mock('react-router-dom', () => {
```
Comment (jroebu14) [inline src/app/containers/App/index.test.jsx]: For greater confidence in our suite of tests, I think we should only mock for tests that need mocks.

Comment (jroebu14) [reply src/app/containers/App/index.test.jsx]: You could revert all changes in this file and do this:

```js
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
}));
```

File: src/app/containers/App/index.test.jsx [hunk 4]
```diff
@@ -28,24 +41,21 @@ describe('ClientApp', () => {
     );
   });
 
+  it('should render App component', () => {
```
Comment (oluoluoxenfree) [inline src/app/containers/App/index.test.jsx]: Why was this added?

Comment (mykeels) [reply src/app/containers/App/index.test.jsx]: It was always there. It used to be on line 46.

File: src/app/containers/App/index.test.jsx (hunk 3 above)
Comment (mykeels) [reply src/app/containers/App/index.test.jsx]: In two tests before, we were mocking `StaticRouter` and `BrowserRouter`, so we could make sure they got called with the right arguments.

The problem is that in `react-router-dom` v5.1.1, `StaticRouter` is now exported with only a get accessor, so we can no longer mock it, the way we were:

```jsx
ReactRouter.StaticRouter = jest.fn(() => <></>);
```

and `jest.spyOn(StaticRouter, ...)` won't work either, because it attempts to set `StaticRouter`.

Instead, using `jest.mock('react-router-dom', ...)`, we can mock the entire module, rather than trying to set `StaticRouter`.

However, we can no longer use jest.fn(() => <></>) as the mocked value of `StaticRouter`, because jest.mock can only be used “globally” to all tests, and other tests depend on the original implementation of `StaticRouter` and `BrowserRouter`.

When I inspected those tests, I saw that they only make sure that the rendered element contains some text.

So, using `jest.fn(({ children})  => <>{children}</a>)` should render the children props, letting the text get rendered, making it a worth replacement mock for both `StaticRouter` and `BrowserRouter`.

Comment (mykeels) [reply src/app/containers/App/index.test.jsx]: An alternative is to separate those two tests, into their own file, where we can `jest.mock(...)` as we please, so the other tests can use the original implementations of `StaticRouter` and `BrowserRouter`.

Comment (mykeels) [reply src/app/containers/App/index.test.jsx]: Following your suggestion on slack, this has been fixed.

**Review by jroebu14 — APPROVED** [review]
Thanks for the changes 👍
